### PR TITLE
Scriban/Liquid support

### PR DESCRIPTION
**Pull request by mholo65**
Basic stuff working. 

- [x] Support for using document as model. `DynamicDocument` does not work (see this https://github.com/lunet-io/scriban/issues/75) so we'd probably need to implement our own `IScriptObject` instead.
- [x] Expose `ParserOptions` and `LexerOptions`. And maybe a simple `.AsLiquid()` to parse template as liquid.
- [x] Support for partials by hooking up Statiq filesystem behind this https://github.com/lunet-io/scriban/blob/master/doc/runtime.md#include-and-itemplateloader

### CODE REVIEW

**Comment by mholo65** [comment]
//cc @xoofx and @daveaglick

**Comment by xoofx** [comment]
> > Just remembered another thing I was wondering: is there a conventional file extension for Scriban templates? I'll probably add this module to Statiq Web twice - once for real Scriban templates and another for Liquid templates with the `.liquid` extension.
> 
> Don't think Scriban has it's own extension. I might be wrong though. Let's ask @xoofx

The extensions that are currently used are listed in the [vscode scriban extension](https://marketplace.visualstudio.com/items?itemName=xoofx.scriban)

File: src/extensions/Statiq.Scriban/StatiqTemplateContext.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                nameof(IDocument.Count),
+                nameof(IDocument.Keys),
+                nameof(IDocument.Values)
+            };
+
+            private readonly IDictionary<IDocument, ImmutableDictionary<string, string>> _documentMetadataCache;
+            private readonly ImmutableDictionary<string, string> _properties;
+            private readonly MemberRenamerDelegate _renamer;
+
+            public DocumentAccessor(MemberFilterDelegate memberFilter, MemberRenamerDelegate memberRenamer)
```
[the remark this answers is not in the document]
Comment (mholo65) [reply src/extensions/Statiq.Scriban/StatiqTemplateContext.cs:38]: On the other hand, I just remove it for now.

**Comment by mholo65** [comment]
@daveaglick everything should be in order now!

**Comment by daveaglick** [comment]
Awesome! Thanks a ton for sticking with it - I'll try to get merged very soon.

**Comment by mholo65** [comment]
Wohoo!

**Comment by daveaglick** [comment]
Looking forward to shipping this one and playing with :) - want to try some purely Scriban themes to see how performance compares to ones with Razor layouts

**Pull request merged**